### PR TITLE
Us032 update readme

### PR DESCRIPTION
**Pull request by TCprof**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,118 @@
...
+
+You will then need to open the JavaWebService file in Intellij which is in:
+
+ Team-A-Agile/backend
+
+Once the project is opened the pom.xml file will run and install the necessary dependencies. If it doesn't then open this file in the editor and reload it by clicking the cycling arrow on the top right.
+
+You will need to create a file called "config.properties" in the resource folder to hold your database credentials. The resource folder is located in:
+
+src/main
```
Comment (KristinaGeddis) [inline README.md:36]:
```suggestion
src/main/resources
```

File: README.md [hunk 2]
```diff
@@ -1 +1,118 @@
...
+
+- [Intellij Download](https://www.jetbrains.com/idea/download/#section=mac)
+
+You will then need to open the JavaWebService file in Intellij which is in:
+
+ Team-A-Agile/backend
+
+Once the project is opened the pom.xml file will run and install the necessary dependencies. If it doesn't then open this file in the editor and reload it by clicking the cycling arrow on the top right.
+
+You will need to create a file called "config.properties" in the resource folder to hold your database credentials. The resource folder is located in:
```
Comment (KristinaGeddis) [inline README.md:34]:
```suggestion
1. You will need to create a file called "config.properties" in the resource folder to hold your database credentials. The resource folder is located in:
```

File: README.md [hunk 3]
```diff
@@ -1 +1,118 @@
...
+Inside config.properties copy/paste the code below and replace the placeholder data with your own.
+
+```
+driver: placeholder
+username: placeholder
+password: placeholder
+url: placeholder
+```
+
+Next you will need to click on the drop-down menu in the top right next to the green hammer and click 'Edit Configurations'
```
Comment (KristinaGeddis) [inline README.md:53]:
```suggestion
2. Next you will need to click on the drop-down menu in the top right next to the green hammer and click 'Edit Configurations'
```

File: README.md [hunk 4]
```diff
@@ -1 +1,118 @@
...
+For the backend you will need to install Intellij Community Edition if you don't already have it. Here is a link to the download page:
+
+- [Intellij Download](https://www.jetbrains.com/idea/download/#section=mac)
+
+You will then need to open the JavaWebService file in Intellij which is in:
+
+ Team-A-Agile/backend
+
+Once the project is opened the pom.xml file will run and install the necessary dependencies. If it doesn't then open this file in the editor and reload it by clicking the cycling arrow on the top right.
+
```
Comment (KristinaGeddis) [inline README.md:33]:
```suggestion
#### Backend set-up steps:
```

File: README.md [hunk 5]
```diff
@@ -1 +1,118 @@
...
+## Setup Instructions
+
+### Backend Setup
+
+####  **Java SDK**
+This project requires Java JDK version 16:
+- [JDK 16 Download](https://www.oracle.com/java/technologies/javase-jdk16-downloads.html)
+
+
+####  **Intellij**
```
Comment (KristinaGeddis) [inline README.md:23]:
```suggestion
###  **Intellij**
```

File: README.md [hunk 6]
```diff
@@ -1 +1,118 @@
...
+        - Microsoft Visual Studio
+- Testing Instructions
+    - Unit Tests
+    - Cucumber Tests
+
+## Setup Instructions
+
+### Backend Setup
+
+####  **Java SDK**
```
Comment (KristinaGeddis) [inline README.md:18]:
```suggestion
###  **Java SDK**
```

File: README.md [hunk 7]
```diff
@@ -1 +1,118 @@
...
+        - Intellij
+    - Frontend Setup
+        - Microsoft Visual Studio
+- Testing Instructions
+    - Unit Tests
+    - Cucumber Tests
+
+## Setup Instructions
+
+### Backend Setup
```
Comment (KristinaGeddis) [inline README.md:16]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -1 +1,118 @@
 # Team-A-Agile
+
+## Table of Contents
+- Setup Instructions
```
Comment (KristinaGeddis) [inline README.md]:
```suggestion
- How to run
    - Run backend
    - Run frontend
- Setup Instructions
```

File: README.md [hunk 9]
```diff
@@ -1 +1,118 @@
...
+    - Backend Setup
+        - Java SDK
+        - Intellij
+    - Frontend Setup
+        - Microsoft Visual Studio
+- Testing Instructions
+    - Unit Tests
+    - Cucumber Tests
+
```
Comment (KristinaGeddis) [inline README.md:13]: Add commands/steps of how to run backend, frontend & what links to view them (http://localhost:8080/api and http://localhost:6555/)

File: README.md [hunk 10]
```diff
@@ -1 +1,118 @@
...
+You should be able to now run the backend by clicking the green arrow in the top right.
+
+---
+
+### Frontend Setup
+
+####  **Microsoft Visual Studio**
+
+For the frontend you will need to install Microsoft Visual Studio Code.
```
Comment (KristinaGeddis) [inline README.md:66]: You don't need VScode, maybe say that's what we recommend using

File: README.md [hunk 11]
```diff
@@ -1 +1,118 @@
...
+You will then need to open the 'frontend' file located in the root directory.
+
+In the 'frontend' folder create a file called ".env". Inside it copy/paste the code below replacing the 'placeholder' with the port number of the backend api. Make sure to save the file once added.
+
+```
+API_URL=http://localhost:placeholder/api
+```
+
+Open the terminal in VScode by going to 'View' -> 'Terminal'
+![reference](images/setup2.png)
```
Comment (KristinaGeddis) [inline README.md:79]: not sure this image is needed

File: README.md [hunk 12]
```diff
@@ -1 +1,118 @@
...
+
+### Frontend Setup
+
+####  **Microsoft Visual Studio**
+
+For the frontend you will need to install Microsoft Visual Studio Code.
+
+- [Microsoft Visual Studio Code download](https://code.visualstudio.com/?wt.mc_id=vscom_downloads)
+
+You will then need to open the 'frontend' file located in the root directory.
```
Comment (KristinaGeddis) [inline README.md:70]:
```suggestion
1. Open the frontend folder
```

File: README.md [hunk 13]
```diff
@@ -1 +1,118 @@
...
+
+In the 'frontend' folder create a file called ".env". Inside it copy/paste the code below replacing the 'placeholder' with the port number of the backend api. Make sure to save the file once added.
```
Comment (KristinaGeddis) [inline README.md:72]:
```suggestion
2. In the root of the frontend folder create a file called ".env". 

+-- frontend
|   +-- .env

3. Inside the .env file, copy the code below replacing the 'placeholder' with the port number of the backend api. Make sure to save the file once added.
```

File: README.md [hunk 14]
```diff
@@ -1 +1,118 @@
...
+
+```
+API_URL=http://localhost:placeholder/api
+```
+
+Open the terminal in VScode by going to 'View' -> 'Terminal'
```
Comment (KristinaGeddis) [inline README.md:78]:
```suggestion
4. Open terminal and type the following command:
```

File: README.md [hunk 15]
```diff
@@ -1 +1,118 @@
...
+![reference](images/setup2.png)
+
+On the command line type "npm install"
+
+Then once the packages have installed type "npm start".
+
```
Comment (KristinaGeddis) [inline README.md]:
```suggestion
```console
npm install
npm start
```

```

File: README.md [hunk 16]
```diff
@@ -1 +1,118 @@
...
+You now should be able to go to the localhost url, replace placeholder with your port number: http://localhost:placeholder/
```
Comment (KristinaGeddis) [inline README.md:91]:
```suggestion
To view the frontend, follow this link: http://localhost:6555/
```

**Review by KristinaGeddis — CHANGES_REQUESTED** [review]

**Review by stephenmcguigan-kainos — COMMENTED** [review]
should you mention that you can use `mvn clean install` to run all the java unit tests at once

File: README.md (hunk 1 above)
Comment (TCprof) [reply README.md:36]: that path is referring to where the resources file is

File: README.md (hunk 15 above)
Comment (TCprof) [reply README.md]: I think its best to go step by step

File: README.md (hunk 14 above)
Comment (TCprof) [reply README.md:95]: should add numbering but will keep the terminal instructions as not everyone will be familiar with the IDE

**Review by stephenmcguigan-kainos — APPROVED** [review]

**Review by AmamMcMam — APPROVED** [review]

**Pull request merged**